### PR TITLE
Added commands for managing delegated funds

### PR DESCRIPTION
**Pull request by popenta**

### CODE REVIEW

File: multiversx_sdk_cli/cli_delegation.py [hunk 1]
```diff
@@ -80,6 +80,41 @@ def setup_parser(args: List[str], subparsers: Any) -> Any:
...
 
+    # delegate
+    sub = cli_shared.add_command_subparser(subparsers, "staking-provider", "delegate",
+                                           "Delegate funds to a delegation contract")
+    sub.add_argument("--delegation-contract", required=True, help="address of the delegation contract")
+    _add_common_arguments(args, sub)
+    sub.set_defaults(func=delegate)
+
+    # claim rewards
+    sub = cli_shared.add_command_subparser(subparsers, "staking-provider", "claim-rewards",
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_delegation.py:91]: :rocket:

File: multiversx_sdk_cli/cli_delegation.py [hunk 2]
```diff
@@ -275,6 +313,82 @@ def unjail_nodes(args: Any):
...
+
+    tx = delegation.prepare_transaction_for_redelegating_rewards(sender, args)
+    cli_shared.send_or_simulate(tx, args)
+
+
+def undelegate(args: Any):
+    cli_shared.check_guardian_and_options_args(args)
+    cli_shared.check_broadcast_args(args)
+    cli_shared.prepare_chain_id_in_args(args)
+    cli_shared.prepare_nonce_in_args(args)
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_delegation.py:365]: A bit of duplication, but can be refactored in another PR (not urgent etc.).

**Review by andreibancioiu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Review by danidrasovean — APPROVED** [review]
